src/main/java/echoserver/EchoServerService.java

```diff
@@ -0,0 +1,33 @@
+package echoserver;
+
+import java.io.IOException;
+import java.net.SocketException;
+import java.util.ArrayList;
+
+public class EchoServerService {
+
+
+    public EchoServerService(HostServer hostServer, Logger logger) {
+        this.hostServer = hostServer;
+        this.logger = logger;
+    }
+
+    public void start() throws IOException {
+        ClientConnection connectedClient;
+
+        while ((connectedClient = hostServer.listenForClientConnection()) != null) {
+            Thread newConnection = new Thread(new EchoServerServiceThread(connectedClient, logger));
```

Comment:
In the domain of echo servers, what is an `EchoServerServiceThread`?

Reply:
Additionally, I would suggest you look at using an Executor vs newing up a thread on each request.

https://docs.oracle.com/javase/tutorial/essential/concurrency/executors.html